torch/csrc/distributed/rpc/tensorpipe_agent.cpp

```diff
@@ -297,15 +322,26 @@ void TensorPipeAgent::startImpl() {
       // a channel that comes before another receives a higher priority.
       priority = opts_.channels->size() - 1 - (iter - opts_.channels->begin());
     }
-    std::unique_ptr<ChannelRegistration> reg =
-        TensorPipeChannelRegistry()->Create(key);
+    std::unique_ptr<CpuChannelRegistration> reg =
+        TensorPipeCpuChannelRegistry()->Create(key);
     if (priority == -1) {
       priority = reg->priority;
     }
     context_->registerChannel(
         priority, std::move(key), std::move(reg->channel));
   }
 
+#if TENSORPIPE_USE_CUDA_IPC_CHANNEL
```

Comment:
```suggestion
#if USE_CUDA
```

Reply:
this means `USE_CUDA` and `TENSORPIPE_HAS_CUDA_IPC_CHANNEL` must always set to the same value?

Reply:
The point of this comment was that the code below must run if there is _any_ CUDA channel, even if it is not exactly the IPC one. Don't you think?